### PR TITLE
[python] Rolling batch support for flash models

### PR DESCRIPTION
**Pull request by xyang16**
## Description ##

Brief description of what this PR is about

- If this change is a backward incompatible change, why must this change be made?
- Interesting edge cases to note here

### CODE REVIEW

File: engines/python/setup/djl_python/rolling_batch/lmi_dist_rolling_batch.py [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+        generation_dict = {}
+        for generation in generations:
+            generation_dict[generation.request_id] = generation
+
+        for r in self.pending_requests:
+            generation = generation_dict[r.id]
+            is_last_token = generation.generated_text is not None
+            r.set_next_token(generation.token_text, last_token=is_last_token)
+
+    def _get_input_ids(self, input_texts):
```
Comment (sindhuvahinis) [inline engines/python/setup/djl_python/rolling_batch/lmi_dist_rolling_batch.py:84]: lets remove this

File: engines/python/setup/djl_python/huggingface.py [hunk 2]
```diff
@@ -121,20 +144,22 @@ def initialize(self, properties: dict):
...
         if self.enable_streaming:
             self._init_model_and_tokenizer(model_id_or_path, **kwargs)
             self.initialized = True
             return
         elif self.enable_rolling_batch:
-            # TODO: Add logic to call appropriate scheduler backend for rolling batch
-            self.rolling_batch = SchedulerRollingBatch(model_id_or_path,
-                                                       self.device, properties,
-                                                       **kwargs)
+            if os.getenv('OMPI_COMM_WORLD_SIZE'):
```
Comment (lanking520) [inline engines/python/setup/djl_python/huggingface.py:154]: check for engine, if engine != Python and tensor_parallel!=1

Comment (xyang16) [reply engines/python/setup/djl_python/huggingface.py]: Changed.

File: engines/python/setup/djl_python/huggingface.py [hunk 3]
```diff
@@ -121,20 +144,22 @@ def initialize(self, properties: dict):
         if "dtype" in properties:
             kwargs["torch_dtype"] = get_torch_dtype_from_str(
                 properties.get("dtype"))
-        self.enable_rolling_batch = properties.get("rolling_batch", None)
-        if self.enable_rolling_batch and self.enable_rolling_batch.lower(
-        ) == "false":
-            self.enable_rolling_batch = None
+        self.enable_rolling_batch = properties.get("rolling_batch") is not None
```
Comment (lanking520) [inline engines/python/setup/djl_python/huggingface.py:147]: self.enable_rolling_batch = properties.get("rolling_batch", None)

File: engines/python/setup/djl_python/huggingface.py [hunk 4]
```diff
@@ -121,20 +144,22 @@ def initialize(self, properties: dict):
...
             self.initialized = True
             return
         elif self.enable_rolling_batch:
-            # TODO: Add logic to call appropriate scheduler backend for rolling batch
-            self.rolling_batch = SchedulerRollingBatch(model_id_or_path,
-                                                       self.device, properties,
-                                                       **kwargs)
+            if self.engine != "Python" and tp_degree != 1:
+                self.device = int(os.getenv("LOCAL_RANK", 0))
+            rolling_batch_type = properties.get("rolling_batch")
```
Comment (lanking520) [inline engines/python/setup/djl_python/huggingface.py:156]: no need for this line

File: engines/python/setup/djl_python/huggingface.py [hunk 5]
```diff
@@ -121,20 +144,22 @@ def initialize(self, properties: dict):
...
+            model_config = AutoConfig.from_pretrained(model_id_or_path, **kwargs)
+            _rolling_batch_cls = get_rolling_batch_class_from_str(rolling_batch_type, model_config)
```
Comment (lanking520) [inline engines/python/setup/djl_python/huggingface.py:158]: _rolling_batch_cls = get_rolling_batch_class_from_str(self.enable_rolling_batch, model_config)

File: engines/python/setup/djl_python/huggingface.py (hunk 4 above)
Comment (xyang16) [reply engines/python/setup/djl_python/huggingface.py]: Done.

File: engines/python/setup/djl_python/huggingface.py (hunk 3 above)
Comment (xyang16) [reply engines/python/setup/djl_python/huggingface.py]: Done

File: engines/python/setup/djl_python/huggingface.py (hunk 5 above)
Comment (xyang16) [reply engines/python/setup/djl_python/huggingface.py]: Done

**Review by lanking520 — APPROVED** [review]

**Pull request merged**